src/rpcmining.cpp

```diff
@@ -606,13 +636,27 @@ Value getblocktemplate(const Array& params, bool fHelp)
     result.push_back(Pair("version", pblock->nVersion));
     result.push_back(Pair("previousblockhash", pblock->hashPrevBlock.GetHex()));
     result.push_back(Pair("transactions", transactions));
-    result.push_back(Pair("coinbaseaux", aux));
-    result.push_back(Pair("coinbasevalue", (int64_t)pblock->vtx[0].vout[0].nValue));
+    if (coinbasetxn) {
+        result.push_back(Pair("coinbasetxn", txCoinbase));
+    } else {
+        result.push_back(Pair("coinbaseaux", aux));
+        // Correct these if GetBlockTemplate changes the order
+        result.push_back(Pair("coinbasevalue", (int64_t)pblock->vtx[0].vout[0].nValue));
+        // Include founders' reward if it is required
+        if (pblock->vtx[0].vout.size() > 1) {
+            result.push_back(Pair("coinbasefrvalue",
+                                  (int64_t)pblock->vtx[0].vout[1].nValue));
+            result.push_back(Pair("coinbasefrscript",
+                                  HexStr(pblock->vtx[0].vout[1].scriptPubKey.begin(),
+                                         pblock->vtx[0].vout[1].scriptPubKey.end())
+                                 ));
+        }
+    }
```

Comment:
"coinbasevalue" in BIP 22 implies the client should create a valid *Bitcoin* transaction paying to *any* scripts. I don't think this will work in zcash, so it should just be "coinbasetxn" until/unless you add some way to negotiate the zcash-specific transaction creation logic.